src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/FilterSpec.java

```diff
@@ -0,0 +1,46 @@
+package io.stargate.sgv2.jsonapi.api.model.command.clause.filter;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.stargate.sgv2.jsonapi.api.model.command.CommandContext;
+import io.stargate.sgv2.jsonapi.api.model.command.deserializers.FilterClauseDeserializer;
+import java.util.Objects;
+
+/**
+ * Intermediate lightly-processed container for JSON that specifies a {@link FilterClause}, as well
+ * as lazily-resolving provider of fully resolved {@link FilterClause}.
+ */
+public class FilterSpec {
+  private static final FilterClauseDeserializer deserializer = new FilterClauseDeserializer();
+
+  /** The JSON value that specifies the filter clause. */
+  private final JsonNode filterJson;
+
+  /** Lazily built POJO representation of the filter clause. */
+  private FilterClause filterClause;
+
+  /**
+   * To deserialize the whole JSON value, need to ensure DELEGATING mode (instead of PROPERTIES).
+   */
+  @JsonCreator(mode = JsonCreator.Mode.DELEGATING)
```

Comment:
Doing this allows as to tell Quarkus we want a `FilterSpec`, and Q (and Jackson) to decode as `JsonNode`, construct light-weight wrapper for it, to be later processed into `FilterClause` when we have `CommandContext` with necessary configuration info (specifically, whether it's Collection or Table that we operate on).